systemtest/src/main/java/io/strimzi/systemtest/bases/IMetrics.java

```diff
@@ -0,0 +1,121 @@
+
+        synchronized (LOCK) {
+            LOGGER.info("Metrics collection for pod {} return code - {}", podName, ret);
+        }
+
+        return exec.out();
+    }
+
+    default HashMap<String, String> collectKafkaPodsMetrics(String clusterName) {
+        LabelSelector kafkaSelector = kubeClient().getStatefulSetSelectors(KafkaResources.kafkaStatefulSetName(clusterName));
+        return collectMetricsFromPods(kafkaSelector);
+    }
+
+    default HashMap<String, String> collectZookeeperPodsMetrics(String clusterName) {
+        LabelSelector zookeeperSelector = kubeClient().getStatefulSetSelectors(KafkaResources.zookeeperStatefulSetName(clusterName));
+        return collectMetricsFromPods(zookeeperSelector);
+    }
+
+    default HashMap<String, String> collectKafkaConnectPodsMetrics(String clusterName) {
+        LabelSelector connectSelector = kubeClient().getDeploymentSelectors(KafkaConnectResources.deploymentName(clusterName));
+        return collectMetricsFromPods(connectSelector);
+    }
+
+    default HashMap<String, String> collectKafkaExporterPodsMetrics(String clusterName) {
+        LabelSelector connectSelector = kubeClient().getDeploymentSelectors(KafkaExporterResources.deploymentName(clusterName));
+        return collectMetricsFromPods(connectSelector, "/metrics");
+    }
+
+
+    /**
+     * Parse out specific metric from whole metrics file
+     * @param pattern regex patern for specific metric
+     * @param data all metrics data
+     * @return list of parsed values
+     */
+    default ArrayList<Double> collectSpecificMetric(Pattern pattern, HashMap<String, String> data) {
+        ArrayList<Double> values = new ArrayList<>();
+
+        data.forEach((k, v) -> {
+            Matcher t = pattern.matcher(v);
+            if (t.find()) {
+                values.add(Double.parseDouble(t.group(1)));
+            }
+        });
+        return values;
+    }
+
+    /**
+     * Collect metrics from all pods with specific selector
+     * @param labelSelector pod selector
+     * @return map with metrics {podName, metrics}
+     */
+    default HashMap<String, String> collectMetricsFromPods(LabelSelector labelSelector) {
+        return collectMetricsFromPods(labelSelector, "");
+    }
+
+    /**
+     * Collect metrics from all pods with specific selector
+     * @param labelSelector pod selector
+     * @param metricsPath additional path where metrics are available
+     * @return map with metrics {podName, metrics}
+     */
+    default HashMap<String, String> collectMetricsFromPods(LabelSelector labelSelector, String metricsPath) {
+        HashMap<String, String> map = new HashMap<>();
+        kubeClient().listPods(labelSelector).forEach(p -> {
+            try {
+                map.put(p.getMetadata().getName(), collectMetrics(p.getMetadata().getName(), metricsPath));
+            } catch (InterruptedException | ExecutionException | IOException e) {
+                e.printStackTrace();
```

Comment:
Much better to rethrow an unchecked exception. It's so easy to miss a stacktrace (now or one day in the future). Better to fail loudly.

```suggestion
                throw new RuntimeException(e);
```